engine/src/main/java/org/terasology/rendering/dag/WireframeTask.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2016 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.rendering.dag;
+
+import org.lwjgl.opengl.GL11;
+import static org.lwjgl.opengl.GL11.GL_FILL;
+import static org.lwjgl.opengl.GL11.GL_FRONT_AND_BACK;
+import static org.lwjgl.opengl.GL11.GL_LINE;
+
+/**
+ *
+ */
+public class WireframeTask implements PipelineTask { // TODO: separate this into SetWireframeOnTask and SetWireframeOffTask?
```

Comment:
Regarding the question in the comment: the tasks pipeline should run as quickly as possible. Therefore, we don't want anything that slows it down, i.e. IF blocks. Two separate classes would remove the need for an IF block in execute().

However, in this particular case, ideally in the constructor you don't store "enabled". You store GL_LINE or GL_FILL depending on the value you get from the StateChange. Then, in execute(), you just need to call:

`GL11.glPolygonMode(GL_FRONT_AND_BACK, mode);`

That removes the need for an IF block and the need for separate Task classes to enable/disable wireframe.